NuGet.cmake

```diff
@@ -25,38 +25,58 @@ include_guard()
 
 include("${CMAKE_CURRENT_LIST_DIR}/ToolchainCommon.cmake")
 
-find_program(NUGET_PATH
-    nuget nuget.exe
-    PATHS
-        ${NUGET_PATH}
-)
+#[[====================================================================================================================
+    toolchain_ensure_nuget
+    ----------------------
 
-# If NuGet isn't found, download it.
-#
-# NuGet.exe will be downloaded to `TOOLCHAIN_TOOLS_PATH`. If `TOOLCHAIN_TOOLS_PATH` is not set then it will
-# be downloaded to '${CMAKE_BINARY_DIR}/__tools'.
-if(NUGET_PATH STREQUAL "NUGET_PATH-NOTFOUND")
-    if(NOT NUGET_VERSION)
-        set(NUGET_VERSION "6.1.0")
-        set(NUGET_HASH "SHA256=432d322b16ffab3fed5d24ab7ea30ae10c555a00339a9bf53060aa4b2aee7925")
-    else()
-        if(NOT NUGET_HASH)
-            message(FATAL_ERROR "NUGET_VERSION is set to ${NUGET_VERSION}. NUGET_HASH must be set if NUGET_VERSION is set.")
+    Ensures that NuGet is available and the NUGET_PATH variables is set in the cache. The following steps will be
+    taken:
+
+        1. If `NUGET_PATH` is set, then that value will be used.
+        2. If NuGet is found through `find_program`, that path will be used.
+        3. NuGet will be downloaded - specified by `NUGET_VERSION` and `NUGET_HASH` - into the `TOOLCHAIN_TOOLS_PATH`
+           and `NUGET_PATH` will be set to the download location. If `TOOLCHAIN_TOOLS_PATH` is not set, then it will
+           default to `${CMAKE_BINARY_DIR}/__tools`.
+
+    Note: Since `CMAKE_BINARY_DIR` is platform specific, the default download location will change by platform,
+    resulting in the tool being downloaded once for each platform that is built. Setting
+    `TOOLCHAIN_TOOLS_PATH` to a platform-independent path (e.g. relative to the root of the repository) will
+    allow NuGet to be downloaded once for all platforms.
+====================================================================================================================]]#
+function(toolchain_ensure_nuget)
+    find_program(NUGET_PATH
+        NAMES nuget nuget.exe
+        PATHS
+            ${NUGET_PATH}
```

Comment:
- Why are you using `PATH ${NUGET_PATH} `and not `HINT`?
- Should `${NUGET_PATH}` the directory were `ninja` is installed or the real path to the `ninja program`?

Reply:
Gosh. You're right. This is all sorts of wrong. I was trying to say that if a value was specified to prefer that, before searching for it. This needs fixing.

Reply:
Thanks for the fixes here. I've rebased this PR on-top of main.